src/BaseModel.js

```diff
@@ -10,23 +10,23 @@ export default class BaseModel {
 	 */
 	constructor(id, createdAt, updatedAt) {
 		/**
-		* The id of the model.
-		*
-		* Unique across all models of the same type.
-		* @type {string}
-		*/
+		 * The id of the model.
+		 *
+		 * Unique across all models of the same type.
+		 * @type {string}
+		 */
 		this.id = id;
 
 		/**
-		* The date that the model was created.
-		* @type {Date}
-		*/
-		this.createdAt = createdAt;
+		 * The date that the model was created.
+		 * @type {Date}
+		 */
+		this.createdAt = createdAt instanceof Date ? createdAt : new Date(createdAt);
 
 		/**
-		* The date that the model was last updated.
-		* @type {Date}
-		*/
-		this.updatedAt = updatedAt;
+		 * The date that the model was last updated.
+		 * @type {Date}
+		 */
+		this.updatedAt = updatedAt instanceof Date ? updatedAt : new Date(updatedAt);
```

Comment:
You can just do `new Date(updatedAt)`, if it's a Date object, it will just clone it